src/gui/fspathedit_p.cpp

```diff
@@ -80,6 +255,33 @@ void Private::FileLineEdit::keyPressEvent(QKeyEvent *e)
         m_completerModel->setRootPath(QFileInfo(text()).absoluteDir().absolutePath());
         showCompletionPopup();
     }
+
+    const FileSystemPathValidator *validator =
+        dynamic_cast<const FileSystemPathValidator *>(this->validator());
```

Comment:
I just thought, why should we use something else, if Qt has it? The only question is effectiveness. I don't know exactly how RTTI complicates it.